toolchain/check/import_cpp.cpp

```diff
@@ -483,15 +497,15 @@ static auto AsCarbonNamespace(Context& context,
                               clang::DeclContext* decl_context)
     -> SemIR::InstId {
   CARBON_CHECK(decl_context);
-  auto& clang_decls = context.sem_ir().clang_decls();
-
-  // Check if the decl context is already mapped to a Carbon namespace.
-  if (auto context_clang_decl_id =
-          clang_decls.Lookup(clang::dyn_cast<clang::Decl>(decl_context));
-      context_clang_decl_id.has_value()) {
-    return clang_decls.Get(context_clang_decl_id).inst_id;
+  // Check if the declaration is already mapped.
```

Comment:
Optional: I think we can avoid this check entirely if we rotate the loops below so they treat the given decl_context the same at its enclosing contexts.

Reply:
TODO removed with #5821.
Hopefully this makes sense.